main.cc

```diff
@@ -1979,6 +1979,9 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 api::unset_server_authorization_cache(ctx).get();
             });
 
+            // update the service level cache after the SL data accessor and auth service are initialized.
+            sl_controller.local().update_cache(qos::update_both_cache_levels::yes).get();
```

Comment:
Oh actually, we don't want to do it unconditional. This should be triggered only if raft topology is enabled and service levels were migrated to v2.

I think that's why dtests failed but dtests with topology changes and with tablets passed. (I've only took a quick look at one random dtest)

Reply:
I changed it to trigger only in v2, thanks.

In the issue I encountered I think what happens is that the legacy loop does an access to the topology state machine, and in the shutdown sequence the tsm is destroyed before the loop is stopped. Maybe it was exposed by some change of timing.